### ISSUE TITLE
BertTokenizer with HuggingFaceModelWrapper

### ISSUE THREAD

**Issue by tsinggggg** [QData/TextAttack#252]
I was trying to attack a customized hugging face model with textattack. When I used the HuggingFaceModelWrapper i got an error about the BertTokenizer.

From https://github.com/QData/TextAttack/blob/master/textattack/models/wrappers/pytorch_model_wrapper.py, the wrapper is expecting the batch_encode method or the encode method from the tokenizer(if the former does not exist)
`    def tokenize(self, inputs):
        if hasattr(self.tokenizer, "batch_encode"):
            return self.tokenizer.batch_encode(inputs)
        else:
            return [self.tokenizer.encode(x) for x in inputs]`

I was using a cusomized transformers.BertTokenizer from someone else which does not seem to have the batch_encode method. Therefore it falls back to the encode method.

However according to https://github.com/huggingface/transformers/blob/63144701ed46d6423b5968db40b6d4469d7d9b87/src/transformers/tokenization_utils_base.py#L1278, encode returns a list of int.

`    def encode(
        self,
        text: Union[TextInput, PreTokenizedInput, EncodedInput],
        text_pair: Optional[Union[TextInput, PreTokenizedInput, EncodedInput]] = None,
        add_special_tokens: bool = True,
        padding: Union[bool, str, PaddingStrategy] = False,
        truncation: Union[bool, str, TruncationStrategy] = False,
        max_length: Optional[int] = None,
        stride: int = 0,
        return_tensors: Optional[Union[str, TensorType]] = None,
        **kwargs
    ) -> List[int]:`

Therefore, the return of the tokenize method from wrapper when there's no batch encode method would be a list of list instead of list of dicts(which are expected by the huggingface wrapper https://github.com/QData/TextAttack/blob/master/textattack/models/wrappers/huggingface_model_wrapper.py#L20), which causes the bug.

I wonder if i did anything wrong? Thank you very much.

**Comment by tsinggggg** [comment]
Sorry! I just realized this issue is an duplicate of #249

**Comment by jxmorris12** [comment]
Hey @tsinggggg -- I literally just updated TextAttack (v0.2.5) to try and fix this. Can you try to `pip install --upgrade` and let me know if it works now?

**Comment by tsinggggg** [comment]
> Hey @tsinggggg -- I literally just updated TextAttack (v0.2.5) to try and fix this. Can you try to `pip install --upgrade` and let me know if it works now?

I am getting another error. I think when tokenizer is a pretrained one, we need to set tokenizer path to None. 

https://github.com/QData/TextAttack/blob/master/textattack/models/wrappers/huggingface_model_wrapper.py#L15

**Comment by jxmorris12** [comment]
My bad. Introduced a new error. That's a poor default. Fixing it @tsinggggg

**Comment by jxmorris12** [comment]
Try again @tsinggggg, I think this should be fixed in v0.2.6, haha.

**Comment by tsinggggg** [comment]
> Try again @tsinggggg, I think this should be fixed in v0.2.6, haha.

Thanks! now I am getting 'forward() has no argument token_type_ids' from a distill bert for sequence classification model...so distill bert and bert do not have the same interface for the forward method. :(

**Comment by jxmorris12** [comment]
Can you share the code you've used? @tsinggggg

**Comment by jxmorris12** [comment]
@tsinggggg make sure your model and tokenizer are being initialized from the same pretrained path

**Comment by tsinggggg** [comment]
> Can you share the code you've used? @tsinggggg

Sorry i was coding on my company laptop so I am not able to share the original code but it's very simple and basically like this. Sorry for any potential typos

import textattack
from transformers import BertTokenizer, DistilBertForSequenceClassification
from textattack.attack_recipes import TextFoolerJin2009

tokenizer = BertTokenizer.from_pretrained(vocab_file_from_someone, do_lower_case=True)
model = DistilBertForSequenceClassification.from_pretrained(model_path_from_someone, num_labels=2)

model_wrapper = textattack.models.wrappers.huggingface_model_wrapper.HuggingFaceModelWrapper(model, tokenizer)

recipe = TextFoolerJin2009.build(model_wrapper)

result = []

for x in recipe.attack_dataset(somedata):
    result.append(x)

**Comment by jxmorris12** [comment]
Thanks @tsinggggg! You need to replace `BertTokenizer` with `DistilBertTokenizer`. Everything else LGTM. DistilBERT uses a different tokenizer than BERT – and Distilbert has no notion of token type IDs, so that's why you saw the error above!

**Comment by jxmorris12** [comment]
@tsinggggg I tried using the `DistilBertTokenizer` and everything worked properly for me! [here's a link to a Colab reproduction](https://colab.research.google.com/drive/1wk2oejNCXpadKvF8G2vMChmSX0n3-ZOD?usp=sharing).
